runtime/vm/jfr.cpp

```diff
@@ -54,13 +56,19 @@ static UDATA collectStackTrace(J9VMThread *currentThread, J9StackWalkState *walk
  *
  * @returns total size in bytes of the event
  */
-static UDATA
+UDATA
 jfrEventSize(J9JFREvent *jfrEvent)
 {
-	UDATA size = sizeof(J9JFREvent);
+	UDATA size = 0;
 	switch(jfrEvent->eventType) {
 	case J9JFR_EVENT_TYPE_EXECUTION_SAMPLE:
-		size += (((J9JFRExecutionSample*)jfrEvent)->stackTraceSize * sizeof(UDATA));
+		size = sizeof(J9JFRExecutionSample) + (((J9JFRExecutionSample*)jfrEvent)->stackTraceSize * sizeof(UDATA));
+		break;
+	case J9JFR_EVENT_TYPE_THREAD_START:
+		size = sizeof(J9JFRThreadStart) + (((J9JFRThreadStart*)jfrEvent)->stackTraceSize * sizeof(UDATA));
+		break;
+	case J9JFR_EVENT_TYPE_THREAD_END:
+		size = sizeof(J9JFREvent);
 		break;
 	default:
```

Comment:
This should assert unreachable - we must not return 0 from here to avoid infinite loops.